src/redis.c

```diff
@@ -61,8 +61,19 @@ redis_meta_get_reply(Meta m)
     int ret;
     redisReply *reply;
 
-    if ((ret = redisGetReply(m->c, (void *)&reply)) == REDIS_OK)
+    if ((ret = redisGetReply(m->c, (void *)&reply)) == REDIS_OK) {
         m->reply = reply;
+    } else {
+        switch (m->c->err) {
```

Comment:
single-case switches make me very sad.

consider this code instead:

    logger_log("%s %d: %s %s", __FILE__, __LINE__, m->c->errstr,
    m->c->err == REDIS_ERR_IO ? strerror(errno) : "");

Sure sometimes the logger will output a spurious space at the end of a line but it's ~1/6th of the code at (at least to me) more readable.